migrations/migrations.go

```diff
@@ -0,0 +1,48 @@
+package migrations
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+)
+
+//Migration Defines a fogg migration and the actions that it can perform
+type Migration interface {
+	Description() string                      //Describes the migration taking
+	Guard(afero.Fs, string) (bool, error)     //Returns true if migration is runnable, otherwise error
+	Migrate(afero.Fs, string) (string, error) //Returns path to config and err
+	Prompt() bool                             //Returns whether the user would like to run the migration
+}
+
+//RunMigrations cycles through a list of migrations and applies them if necessary
+func RunMigrations(fs afero.Fs, configFile string, skipAll bool) error {
+	migrations := []Migration{
+		&VersionUpgradeMigration{}, //This runs first because v1 to v2 uses json.Marshal
+		&JSONToYamlMigration{},
+	}
+
+	for _, migration := range migrations {
+		shouldRun, err := migration.Guard(fs, configFile)
+		if err != nil {
+			return err
+		}
+		if shouldRun == false {
```

Comment:
maybe this could use a `log.Debug()` or something to give feedback that you've skipped a migration